src/Processors/Formats/Impl/CSVRowInputFormat.cpp

```diff
@@ -561,7 +563,11 @@ std::pair<bool, size_t> fileSegmentationEngineCSVImpl(ReadBuffer & in, DB::Memor
             else if (*pos == '\r')
             {
                 ++pos;
-                if (loadAtPosition(in, memory, pos) && *pos == '\n')
+                const DB::ContextPtr context = DB::CurrentThread::get().getGlobalContext();
+                auto format_settings = getFormatSettings(context);
```

Comment:
We should not use `DB::CurrentThread::get().getGlobalContext()`. We should propogate query context to this place. And it will require some changes in FormatFactory.
We can change `file_segmentation_engine` here:
https://github.com/ClickHouse/ClickHouse/blob/e71ca7dcefe11f1e37669b31c68426979b44961f/src/Formats/FormatFactory.h#L135
To `file_segmentation_engine_creator` with type `FileSegmentationEngine = std::function<FileSegmentationEngine(const FormatSettings & settings)>`
And add new method to FormatFactory:
```c++
registerFileSegmentationEngineCreator(const String & name, FileSegmentationEngineCreator file_segmentation_engine);
```
And we will need to change `registerFileSegmentationEngine` implementation so it uses `file_segmentation_engine_creator`.

After doing it, we can use `registerFileSegmentationEngineCreator` in CSV format, so we will be able to use `FormatSettings` to propogate our setting.

Reply:
hello, I have modify the code according to you advices, but the test still can not pass, please help to review my code if the implemetion is right ? @Avogar